src/gluonts/dataset/parallelized_loader.py

```diff
@@ -346,6 +355,45 @@ def _worker_fn(
     return buf.getvalue()
 
 
+class _ShuffleBatchIter(Iterator[DataEntry]):
+    def __init__(
+        self,
+        base_iterator: Iterator[DataEntry],
+        shuffle_buffer_length: Optional[int],
+        multiprocess: bool,
+    ):
+        if multiprocess:
+            self.shuffle_buffer = _WorkerData.shuffle_buffer
+        else:
+            self.shuffle_buffer = []
+        self.shuffle_buffer_length = shuffle_buffer_length
+        self.base_iterator = base_iterator
+
+    def __next__(self) -> DataEntry:
+        try:
+            if self.shuffle_buffer_length is not None:
+                if len(self.shuffle_buffer) == 0:
+                    for element in itertools.islice(
+                        self.base_iterator, self.shuffle_buffer_length
+                    ):
+                        self.shuffle_buffer.append(element)
+                idx = random.randint(0, self.shuffle_buffer_length - 1)
+                next_sample = self.shuffle_buffer[idx]
+                self.shuffle_buffer[idx] = next(self.base_iterator)
+                return next_sample
+            else:
+                return next(self.base_iterator)
+        except StopIteration:
+            return {}
```

Comment:
Why is this needed? Without this you could remove the try...except

Reply:
Previously I am using next(self.base_iterator) so StopIteration Exception may occur. Now I've  edited the way I've written and removed the try/except.